packages/devtools/lib/src/tables.dart

```diff
@@ -209,24 +229,29 @@ class Table<T> extends Object with SetStateMixin {
       }
     }
 
-    data.sort((T a, T b) {
-      if (numeric) {
-        final num one = column.getValue(a);
-        final num two = column.getValue(b);
-        if (one == two) {
-          return 0;
-        }
-        if (_sortDirection == SortOrder.ascending) {
-          return one > two ? 1 : -1;
-        } else {
-          return one > two ? -1 : 1;
-        }
+    _sortData(column, direction);
+  }
+
+  void _sortData(Column column, int direction) =>
+      data.sort((T a, T b) => _compareData(a, b, column, direction));
+
+  int _compareData(T a, T b, Column column, int direction) {
+    if (column.numeric) {
+      final num one = column.getValue(a);
+      final num two = column.getValue(b);
+      if (one == two) {
+        return 0;
+      }
+      if (_sortDirection == SortOrder.ascending) {
+        return one > two ? 1 : -1;
       } else {
-        final String one = column.render(column.getValue(a));
-        final String two = column.render(column.getValue(b));
-        return one.compareTo(two) * direction;
+        return one > two ? -1 : 1;
       }
-    });
+    } else {
+      final String one = column.render(column.getValue(a));
+      final String two = column.render(column.getValue(b));
+      return one.compareTo(two) * direction;
```

Comment:
note this else case would actually work for the num case as well. You can make the type on the LHS Comparable for one and two and everything will just work.